File: app/src/main/java/com/mnj/java_solutions/linkedlist/MiddleOfLinkedList.java
package com.mnj.java_solutions.linkedlist;

public class MiddleOfLinkedList {

    public void middleNode() {
        ListNode node = new ListNode();
        ListNode.Node head = node.insert();

        ListNode.Node fast = head;
        ListNode.Node slow = head;

        while (fast != null && fast.next != null) {
            fast = fast.next.next;
            slow = slow.next;
        }
        System.out.println("==>> Middle of linked list is :" + slow.data);
    }
}

File: app/src/main/java/com/mnj/java_solutions/linkedlist/MiddleOfLinkedList.kt
package com.mnj.java_solutions.linkedlist

import com.mnj.java_solutions.linkedlist.ListNode

class MiddleOfLinkedList {
    fun middleNode() {
        val node = ListNode()
        val head = node.insert()
        var fast = head
        var slow = head
        while (fast?.next != null) {
            fast = fast.next.next
            slow = slow.next
        }
        println("==>> Middle of linked list is :" + slow.data)
    }
}